### ISSUE TITLE
Hover displaying datetime using wrong locale

### ISSUE THREAD

**Issue by verybadsoldier** [bokeh/bokeh#7408]
*Disclaimer*: I am new to Bokeh but I like it alot! Please consider that I am just doing something wrong ;)

When using datetime objects within a hover then in my understanding it should be displayed using the current locale settings when using format modifier "%c".
 According to: https://bokeh.pydata.org/en/latest/docs/reference/models/formatters.html#bokeh.models.formatters.DatetimeTickFormatter.

So as I am in germany and locale is german, time should be displayed as for example "21:20:37" (no AM/PM). Instead it uses a US format (AM/PM):
![image](https://user-images.githubusercontent.com/364469/34908289-77c13b76-f88d-11e7-87dd-bb60d958b9de.png)

As an example I modified the candlestick example to show a hover:
```
from math import pi
from bokeh.models import HoverTool

import pandas as pd

from bokeh.plotting import figure, show, output_file
from bokeh.sampledata.stocks import MSFT

df = pd.DataFrame(MSFT)[:50]
df["date"] = pd.to_datetime(df["date"])

inc = df.close > df.open
dec = df.open > df.close
w = 12*60*60*1000 # half day in ms

TOOLS = "pan,wheel_zoom,box_zoom,reset,save"

p = figure(x_axis_type="datetime", tools=TOOLS, plot_width=1000, title = "MSFT Candlestick")
p.xaxis.major_label_orientation = pi/4
p.grid.grid_line_alpha=0.3

h = HoverTool(tooltips=[('Time', '@date{%c}')],
              mode="vline",
              formatters={'date': 'datetime'}
              )
p.tools.append(h)

p.segment(df.date, df.high, df.date, df.low, color="black")
p.vbar('date', w, 'open', 'close', fill_color="#D5E1DD", line_color="black", source=df)

output_file("candlestick.html", title="candlestick.py example")

show(p)  # open a browser
```

*Environment:*
Windows 10
Python 3.6.3 Anaconda
Bokeh 0.12.13
Google Chrome

So do I manually have to set the locale I want to use? In my understanding it should be auto detected? Thanks for your time and for Bokeh!

**Comment by bryevdv** [comment]
> So do I manually have to set the locale I want to use? In my understanding it should be auto detected?

I think this is a documentation error. After various false starts and problems, the ultimate solution we are able to maintain is simply to make BokehJS timezone-naive. Or in other words, datetimes are always displayed "as-is" by Bokeh built-ins. It would be possible to make custom models or formatters that has different assumptions and does whatever timezone manipulations you need. 

I'll leave this issue open as a docs task.

**Comment by bryevdv** [comment]
Edit: looking a the actual docs line in question, I am not sure it is an error, per-se. The locale has to do with how the time is presented (e.g 24hr vs am/pm) but not anything to to do with timezones. But we can make that more clear with a note.

**Comment by verybadsoldier** [comment]
Yes, sorry if I was unclear in my example: in my opinion this error is not about timezones. The issue is solely about how the time is displayed (exactly as you said in your previous post).

So what can I do to have the time displayed as "14:12" instead of "2:12 PM"? I thought it would automatically do it right (hopefully) according to my locale.

**Comment by bryevdv** [comment]
Oh I see, we will have to investigate what the underlying third-party library that BokehJS uses actually does. Offhand I would assume that it would check some sort of browser setting that is exposed to the JS runtime, but I am not certain. I would no expect for a JS lib to be able to access any global system locale setting, unless the browser "forwards" that information to the JS sandbox.

**Comment by bryevdv** [comment]
Some possible information here:

https://stackoverflow.com/questions/673905/best-way-to-determine-users-locale-within-browser

Tho at a glance I am not sure it is promising wrt to what you'd like to see happen.

**Comment by mattpap** [comment]
There's `Intl` module in the standard library that could help with this (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DateTimeFormat).

**Comment by bryevdv** [comment]
Right I don't think the actual formatting is an issue, as much as reliably getting the right locale to use since evidently normal practice would have it be in an HTTP header

**Comment by mattpap** [comment]
`Intl` supports auto-detection. You can check the results with `Intl.DateTimeFormat().resolvedOptions()`. Though I'm not sure how accurate this is. On my system, which is a mixture of `en-US` and `pl-PL`, with the former being the main locale, it correctly detects this.
